Source/Schema.NET/OneOrMany{T}.cs

```diff
@@ -15,131 +16,130 @@ namespace Schema.NET
                 {
-                    items.Add(itemT);
+                    for (var i = 0; i < array.Length; i++)
+                    {
+                        var item = array[i];
+                        if (!string.IsNullOrWhiteSpace(item as string))
+                        {
+                            items[index] = item;
+                            index++;
+                        }
+                    }
+                }
+                else
+                {
+                    for (var i = 0; i < array.Length; i++)
+                    {
+                        var item = array[i];
+                        if (item != null)
+                        {
+                            items[index] = item;
+                            index++;
+                        }
+                    }
                 }
-            }
 
-            if (items.Count == 1)
-            {
-                this.collection = null;
-                this.item = items[0];
-            }
-            else
-            {
-                this.collection = items;
-                this.item = default;
+                if (index > 0)
+                {
+                    if (index == array.Length)
+                    {
+                        this.collection = items;
+                    }
+                    else
+                    {
+                        this.collection = new T[index];
+                        Array.Copy(items, 0, this.collection, 0, index);
+                    }
+
+                    this.HasOne = index == 1;
+                    return;
+                }
             }
+
+            this.collection = null;
+            this.HasOne = false;
         }
 
         /// <summary>
-        /// Gets the number of elements contained in the <see cref="OneOrMany{T}"/>.
+        /// Initializes a new instance of the <see cref="OneOrMany{T}"/> struct.
         /// </summary>
-        public int Count
+        /// <param name="collection">The collection of values.</param>
+        public OneOrMany(IEnumerable<T> collection)
+            : this(collection.ToArray())
         {
-            get
-            {
-                if (this.HasOne)
-                {
-                    return 1;
-                }
-                else if (this.HasMany)
-                {
-                    return this.collection.Count;
-                }
+        }
 
-                return 0;
-            }
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OneOrMany{T}"/> struct.
+        /// </summary>
+        /// <param name="collection">The list of values.</param>
+        public OneOrMany(IEnumerable<object> collection)
+            : this(collection?.Cast<T>())
+        {
         }
 
+        /// <summary>
+        /// Gets the number of elements contained in the <see cref="OneOrMany{T}"/>.
+        /// </summary>
+        public int Count => this.collection?.Length ?? 0;

```

Comment:
I thought about doing this. Perhaps we should also implement `ICollection` if we've come this far?

Reply:
Yeah, that's a good idea.

Reply:
Going back over this - we already implement `IReadOnlyCollection<T>` and as `OneOrMany` is immutable, many over the implemented interface methods on `ICollection<T>` would be no-ops or need to throw exceptions.

Reply:
Yes I agree about `ICollection<T>`. I think we could do `ICollection` for `CopyTo` but not a massive improvement.

Reply:
`CopyTo` might be interesting - will look into it.